web/src/search/results/SearchResults.tsx

```diff
@@ -107,12 +116,15 @@ export class SearchResults extends React.Component<SearchResultsProps, SearchRes
                     map(props => [
                         parseSearchURLQuery(props.location.search, props.interactiveSearchMode),
                         parseSearchURLPatternType(props.location.search),
+                        parseSearchURLCaseSensitive(props.location.search),
                     ]),
                     // Search when a new search query was specified in the URL
                     distinctUntilChanged((a, b) => isEqual(a, b)),
                     filter(
-                        (queryAndPatternType): queryAndPatternType is [string, GQL.SearchPatternType] =>
-                            !!queryAndPatternType[0] && !!queryAndPatternType[1]
+                        (
+                            queryAndPatternTypeAndCase
+                        ): queryAndPatternTypeAndCase is [string, GQL.SearchPatternType, boolean] =>
+                            !!queryAndPatternTypeAndCase[0] && !!queryAndPatternTypeAndCase[1]
```

Comment:
Should the third element also be checked so that the typeguard doesn't 'lie'? Also, I feel like queryAndPatternTypeAndCase would be better represented as a plain object rather than a tuple of three elements.

```suggestion
                            !!queryAndPatternTypeAndCase[0] && !!queryAndPatternTypeAndCase[1] && typeof queryAndPatternTypeAndCase[2] === 'boolean'
```